docs/design-docs/02-custom-root-reconciler-clusterrole.md

```diff
@@ -0,0 +1,133 @@
+
+## Design Overview
+
+By providing the name of a user-defined `ClusterRole` for a `RootSync`, a user can
+override which role Config Sync binds to. This configuration is exposed as a new field
+on `spec.overrides` for a `RootSync`:
+
+```yaml
+spec:
+  overrides:
+    clusterRole:
+      name: my-custom-role
+```
+
+This role is defaulted to `cluster-admin` in order to stay backwards compatible.
+
+For the case where a single `ClusterRole` is not expressive enough to configure the
+permissions a user want, you can instead set `disabled: true` on the override object:
+
+```yaml
+spec:
+  overrides:
+    clusterRole:
+      disabled: true
+```
+This disables creating the `ClusterRoleBinding` entirely.
+
+If both `disabled: true` and a custom name is specified, `disabled` "wins" and no binding
+is created.
+
+## User Guide
+
+To take advantage of this new feature, set `spec.overrides.clusterRole` to the name of a
+`ClusterRole` you wish this `RootSync` to be reconciled under; you must create this role
+yourself, but Config Sync will create the `ClusterRoleBinding` for you.
+
+```yaml
+apiVersion: configsync.gke.io/v1beta1
+kind: RootSync
+metadata:
+  name: root-sync
+  namespace: config-management-system
+spec:
+  overrides:
+    clusterRole:
+      name: my-cluster-role
+  # ...
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: my-cluster-role
+rules:
+# ...
+```
+
+You can also disable automatic binding to any role.
+
+```yaml
+apiVersion: configsync.gke.io/v1beta1
+kind: RootSync
+metadata:
+  name: root-sync
+  namespace: config-management-system
+spec:
+  overrides:
+    clusterRole:
+      disabled: true
+  # ...
+```
+
+The reconciler will then run with effectively no permissions, until you manually create
+some role or cluster role bindings for it.
+
+**Note** that while the name of the relevant service account will _often_ be predictable
+as `root-reconciler-<root-reconciler-name>`, this is not guaranteed by Config Sync. You
+can always find the service account by its labels; in particular, the labels
+`configsync.gke.io/sync-kind` and `configsync.gke.io/sync-name` should be useful.
+
+## Risks and Mitigations
+
+This feature is entirely backwards-compatible, by means of being opt-in.
+
+## Test Plan
+
+Implementation of this feature can includes automated tests that verify the behavior.
+
+## Open Issues/Questions
+
+N/A
+
+## Alternatives Considered
+
+One could imagine exposing settings with slightly different semantics - e.g. a simple
+boolean for turning the `ClusterRoleBinding` off, or a setting to change what service
+account the reconciler is running with. However, these both put a bigger burden on the
+user in order to utilize them even for the simple use cases, which is why changing which
+role to bind to is probably the most user-friendly knob to expose. With the inclusion of
+a sentinel value to create no binding, the first of these alternatives is effectively
```

Comment:
Please update this snippet to reflect that the sentinel value is no longer being used

Reply:
Done.